pkg/controller/podautoscaler/horizontal.go

```diff
@@ -453,7 +453,12 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 		case desiredReplicas == 0:
 			//  never scale down to 0, reserved for disabling autoscaling
 			setCondition(hpa, autoscalingv2.ScalingLimited, v1.ConditionTrue, "TooFewReplicas", "the desired replica count was zero")
-			desiredReplicas = 1
+			// make sure we aren't below our minimum if one is defined
+			if hpa.Spec.MinReplicas != nil {
+				desiredReplicas = *hpa.Spec.MinReplicas
```

Comment:
Sorry, I'm not sure I understand? In case of near zero load (which causes the desired replicas to be calculated as 0), the hpa should still try to downscale to a possible minimum.

This PR changes this behaviour only to scale to the defined minimum instead of hardcoded `1`, which causes all except one running pod to be killed and the HPA scaling to the defined min during the next iteration with `Current number of replicas below Spec.MinReplicas`